src/components/DayPicker.jsx

```diff
@@ -68,11 +72,17 @@ const defaultProps = {
   monthFormat: 'MMMM YYYY',
 };
 
+const triggerInitialVisibleMonth = (props) => (
+  props.hidden ? moment() : props.initialVisibleMonth()
```

Comment:
I'm not sure that we need this logic to be perfectly honest. The constructor is only called once and even if the datepicker is visible at that point, it will be the first time its ever rendered so I think we can just set `state.currentMonth` to `props.initialVisibleMonth()`. What is the edge case this is trying to solve?